lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -1138,6 +1216,93 @@ fixtureMultiAssetWallet
     -> ResourceT m ApiWallet
 fixtureMultiAssetWallet = fmap fst . fixtureWalletWithMnemonics (Proxy @"ma")
 
+fixtureMultiAssetRandomWallet
+    :: forall n m.
+        ( DecodeAddress n
+        , DecodeStakeAddress n
+        , EncodeAddress n
+        , MonadIO m
+        , MonadUnliftIO m
+        )
+    => Context
+    -> ResourceT m ApiByronWallet
+fixtureMultiAssetRandomWallet ctx = do
+    wMA <- fixtureMultiAssetWallet ctx
+    wB <- fixtureRandomWallet ctx
+
+    -- create Byron address
+    let p = Json [aesonQQ| { "passphrase": #{fixturePassphrase} }|]
+    r <- request @(ApiAddress n) ctx (Link.postRandomAddress wB) Default p
+    expectSuccess r
+
+    -- pick out assets to send
```

Comment:
It would be nice if the shelley wallet migration endpoint were available to use here...